store/src/test/java/com/dropbox/android/external/store4/impl/ClearAllStoreTest.kt

```diff
@@ -0,0 +1,93 @@
+package com.dropbox.android.external.store4.impl
+
+import com.dropbox.android.external.store4.ExperimentalStoreApi
+import com.dropbox.android.external.store4.StoreBuilder
+import com.dropbox.android.external.store4.get
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import java.util.concurrent.atomic.AtomicInteger
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+@ExperimentalStoreApi
+@RunWith(JUnit4::class)
+class ClearAllStoreTest {
+
+    private val testScope = TestCoroutineScope()
+
+    private val key = "key"
+
+    private val networkCalls = AtomicInteger(0)
+    private val deleteAllCalls = AtomicInteger(0)
+
+    private val diskReader = mock<(String) -> Flow<Int?>>()
+    private val diskWriter = mock<suspend (String, Int) -> Unit>()
+
+    @Test
+    fun `clearAll() with persister`() = testScope.runBlockingTest {
```

Comment:
I like to add to the test name what am I testing, not just what scenario I'm running under (though I admit it makes for long test names)